### DISCUSSION TITLE
Automated replacement of emotes

### DISCUSSION THREAD

**Discussion by ChristianR666** [Chatterino/chatterino2#3621] [Ideas]
I want to replace the LUL emote with another emote of the channel. I tried it by using the commands.
Trigger: LUL
Command: {1+} 'new emote'

First problem: if I write something in front of the trigger, the command doesn´t trigger..
Input: test LUL
![grafik](https://user-images.githubusercontent.com/102114799/159381813-5624c09b-f55d-47cc-bc68-8d7674546bd1.png)

Second problem: if I write something after the trigger the text is set in front of the new emote..
Input: LUL test
![grafik](https://user-images.githubusercontent.com/102114799/159381907-25dd6ccb-f057-4e66-a789-eb458a299ad8.png)

My idea: a setting to replace the emote you want and set an alias for it, so the text before and after doesn´t change or will be placed somewhere else in the sentence. Also there should be an option to choose if upper- or lowercase letters to be considered.
Example: LUL --> 'new emote'
lul --> 'lul'
something with 'lul' in the word --> that word without an emote in middle of the word

**Comment by LosFarmosCTL** [comment]
So if I understand this correctly, you would like to have a feature like ignores, but for outgoing messages you send yourself?

**Reply by Felanbird** [reply]
They want the functionality of commands firing when not used at the start of a message, as well as the usage of the command to be fired despite the location of the emote in the message `{^+} emote_here {$+}` (if it was possible to know exactly how many words you were going to type `{1+} emote_here {16+}`)

**Reply by LosFarmosCTL** [reply]
Sounds more like a seperate issue/request to me that he tried solving with commands, which are probably just completely unsuited for what he wanted to do.

**Reply by Felanbird** [reply]
> Sounds more like a seperate issue/request to me that he tried solving with commands, which are probably just completely unsuited for what he wanted to do.

Ah fair, looking at it that way makes more sense, it would require some sort of text-replace setting client side since I assume it's intended to "use" LUL but send the sub emote anyway.

**Comment by Felanbird** [comment]
> Second problem: if I write something after the trigger the text is set in front of the new emote..

This is intentional behavior because your command was `{+1} emote_here`, if you want the text after the emote you want `emote_here {1+}`.
